src/lib/sse/EventSourceError.ts

```diff
@@ -0,0 +1,26 @@
+interface EventSourceInfo {
+  url: string
+  readyState: number
+  eventsCount: number
+  errorCount: number
+  pingEventsCount: number
+  restartCount: number
+  config: any
+}
+
+export class EventSourceError extends Error {
+  public event: any
+  public eventSourceInfo: EventSourceInfo
+  constructor(event: any, eventSourceInfo: EventSourceInfo) {
+    super(`SSE error on endpoint ${eventSourceInfo.url}`)
+    this.eventSourceInfo = eventSourceInfo
+    this.event = { ...event }
```

Comment:
Why copying the event?

Reply:
This was because of an error I got - the `event` object created by the `eventsource` lib has some write protections. This created an error when calling `ErrorReport.toObject` because when truncating the strings from the object at some point we do `object[key] = ...` (this is forbidden because of the write protection)

One way to do this write protection for example is using: https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

Reply:
Nice!!